jib-core/src/main/java/com/google/cloud/tools/jib/docker/DockerClient.java

```diff
@@ -61,13 +76,19 @@ public static DockerClient newClient(Path dockerExecutable) {
    * @param dockerExecutable path to {@code docker}
    * @return the default {@link ProcessBuilder} factory for running a {@code docker} subcommand
    */
-  private static Function<List<String>, ProcessBuilder> defaultProcessBuilderFactory(
-      String dockerExecutable) {
+  @VisibleForTesting
+  static Function<List<String>, ProcessBuilder> defaultProcessBuilderFactory(
```

Comment:
We probably want to make `dockerEnvironment` immutable so there's no expected behavior if the user changes the map they passed into the builder since the map can change and the environment used by the `ProcessBuilder` would be different.

Reply:
Ah, good catch. I think we can copy the map at the beginning.